packages/stryker/test/unit/TempFolderSpec.ts

```diff
@@ -0,0 +1,84 @@
+import { TempFolder } from '../../src/utils/TempFolder';
+
+import * as sinon from 'sinon';
+import { expect } from 'chai';
+import * as mkdirp from 'mkdirp';
+import * as fs from 'mz/fs';
+
+describe.only('TempFolder', () => {
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+  afterEach(() => sandbox.restore());
+
+  it('TempFolder is presented', () => {
+    expect(TempFolder).not.to.be.undefined;
+
+    const tempFolderInstance = TempFolder.instance();
+
+    expect(tempFolderInstance.createRandomFolder).to.exist;
+    expect(tempFolderInstance.writeFile).to.exist;
+    expect(tempFolderInstance.copyFile).to.exist;
+    expect(tempFolderInstance.clean).to.exist;
+    expect(tempFolderInstance.random).to.exist;
+  });
+
+  describe('createRandomFolder', () => {
+    it('should create dir with correct path', () => {
+      const mockCwd = process.cwd() + '/some/dir';
+      sandbox.stub(mkdirp, 'sync');
+      sandbox.stub(process, 'cwd').returns(mockCwd);
+      sandbox.stub(TempFolder.instance(), 'random').returns('rand');
+
+      const result = TempFolder.instance().createRandomFolder('prefix');
+
+      // expect(mkdirp.sync).to.have.been.calledThrice;
+      expect(mkdirp.sync).to.have.been.calledOnce; // only once since I added test which also calls instance 
+      expect(result.includes('prefix')).to.be.true;
+      expect(result.includes('rand')).to.be.true;
+    });
+  });
+
+  describe('clean', () => {
+    it('should call deleteDir fileApi', () => {
+      // TODO
+    });
+  });
+
+  describe('copyFile', () => {
+    it('should call fs api', () => {
+      // TODO
+    });
+  });
+
+  describe('writeFile', () => {
+    beforeEach(() => {
+      sandbox.stub(fs, 'writeFile');
+    });
+    it('should call fs.writeFile', () => {
+      TempFolder.instance().writeFile('filename', 'data');
+      expect(fs.writeFile).to.have.been.calledWith('filename', 'data', {
+        encoding: 'utf8'
+      });
+    });
+  });
+
+  xdescribe('constructor logic', () => {
```

Comment:
Should this still be removed? Or changed into the testing of `initialize()`?

Reply:
Yes it should be changed. Tests are not ready. I will update them since current code is ok (as I am getting from your review). I just wanted to know if TempFolder.ts code is correct.

Reply:
I removed it.